### ISSUE TITLE
Slicing issue

### ISSUE THREAD

**Issue by brstra** [percipient/django-querysetsequence#23]
Hello,
I've found something weird.

```
objs = Objs.objects.all()
length = len(objs[1:4])
# length equals 3

objs = QuerySetSequence(Objs.objects.all())
length = len(objs[1:4])
# length equals 4
```

Actually, it looks like QuerySetSequence ignores start index and it always equals 0.

As result, it breaks Django pagination. I found a workaround for that - convert QSS to a list, and pass that list to Paginator class. But maybe it should not be like that?

**Comment by clokep** [comment]
Interesting, this could be a regression from some of the pagination things we've added recently.

Any chance you can make a test-case in the testing framework to show this failing? I've tried to reproduce it and haven't been able to.

Also, are you using the latest version (0.7)?

**Comment by brstra** [comment]
Aha, ok. I'll test it in a clean environment. 
I used the last one from PyPI.

**Comment by clokep** [comment]
I had a thought last night...maybe there's an issue when using only a single `QuerySet` inside of the `QuerySetSequence`, I'll admit that I haven't done much testing in that case (since it's somewhat degenerative).

**Comment by brstra** [comment]
Hey again. Was really busy, but I've got some time today to do some testing. I've made clean install (Python 3.5.2, Django 1.10.6, freshly pip-installed django-querysetsequence). 

Here some funny results:
```
class NoteTests(TestCase):
    def setUp(self):
        # Note model is primitive and contains only one field 'text' VARCHAR(255)
        Note.objects.bulk_create([
            Note(text="Note number 1"),
            Note(text="Note number 2"),
            Note(text="Note number 3"),
            Note(text="Note number 4"),
            Note(text="Note number 5"),
            Note(text="Note number 6")
        ])
    def test_opened_slice(self):
        # This one works
        notes = Note.objects.all()
        qs_notes = notes[1:]
        qss_notes = QuerySetSequence(notes)[1:]

        self.assertEqual(len(qs_notes), len(qss_notes))

    def test_closed_slice_single_qs(self):
        # This one failes
        notes = Note.objects.all()
        qs_notes = notes[1:4]
        qss_notes = QuerySetSequence(notes, notes)[1:4]

        self.assertEqual(len(qs_notes), len(qss_notes))

    def test_closed_slice_two_qs_filter(self):
        # This one works
        notes_1 = Note.objects.filter(pk__lte=2)
        notes_2 = Note.objects.filter(pk__gte=1)

        qss_concat = QuerySetSequence(notes_1, notes_2)

        self.assertNotEqual(len(qss_concat[0:4]), len(qss_concat[2:4]))

    def test_closed_slice_two_qs_all(self):
        # This one fails
        notes_1 = Note.objects.all()
        notes_2 = Note.objects.all()

        qss_concat = QuerySetSequence(notes_1, notes_2)

        self.assertNotEqual(len(qss_concat[0:4]), len(qss_concat[2:4]))
```
Fail messages:
```
======================================================================
FAIL: test_closed_slice_single_qs (modules.homepage.tests.NoteTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pr/modules/homepage/tests.py", line 29, in test_closed_slice_single_qs
    self.assertEqual(len(qs_notes), len(qss_notes))
AssertionError: 2 != 4

======================================================================
FAIL: test_closed_slice_two_qs_all (modules.homepage.tests.NoteTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pr/modules/homepage/tests.py", line 45, in test_closed_slice_two_qs_all
    self.assertNotEqual(len(qss_concat[0:4]), len(qss_concat[2:4]))
AssertionError: 4 == 4

----------------------------------------------------------------------
```

**Comment by clokep** [comment]
Thanks for providing tests cases! I think I was able to figure this out. I think it's a bug in the handling when the slicing brings us down to a *single* queryset. I'm going to put up a PR in a bit.

**Comment by clokep** [comment]
I just pushed a fix for this in `0.7.1`, let me know if you're still seeing issues! Thanks for reporting. :)
